Compiling normaliz 3.6.0 with flint2 installed leads to error

On a machine with flint 2.4.5 installed, compiling normaliz 3.6.0 runs into a compile error:
```
$ ./configur && make
...
In file included from libnormaliz/enumeration.cpp:28:
./libnormaliz/HilbertSeries.cpp:226:9: error: use of undeclared identifier 'fmpz_poly_cyclotomic'
        fmpz_poly_cyclotomic(cyc, (ulong) n);
        ^
1 error generated.
```

I am told this is because my version of flint is too old. That's fine -- but I think that the configure script's check for flint then should either check the version of flint, and/or check for the availability of that symbol, at least when it is auto-detecting flint (i.e. the user did not give an explicit `--with-flint`); and it then should *not* use flint if the found flint version is not suitable.

Also, while I am at it: I think it'd be helpful if the configure script printed something like "checking for flint ... yes / no", which IMHO is a best practice for configure scripts.

--- a/.travis-build.sh
+++ b/.travis-build.sh
@@ -118,7 +118,7 @@ case $BUILDSYSTEM in
     *)
 	# autotools, no Flint
 	./bootstrap.sh || exit 1
-	./configure $CONFIGURE_FLAGS --prefix="$INSTALLDIR" --disable-flint || ( echo '#### Contents of config.log: ####'; cat config.log; exit 1)
+	./configure $CONFIGURE_FLAGS --prefix="$INSTALLDIR" --with-flint=no || ( echo '#### Contents of config.log: ####'; cat config.log; exit 1)
 
 	make -j2 -k || exit 1
 	make -j2 -k check || exit 1

--- a/configure.ac
+++ b/configure.ac
@@ -206,32 +206,31 @@ AS_IF([test enable_nmzintegrate = xyes],
 
 ### Test whether to use flint
 AC_ARG_WITH([flint],
-   AS_HELP_STRING([--with-flint=DIR],
-      [provide location of flint installation prefix or source directory]),
-   [ LDFLAGS="-L$with_flint/lib -Wl,-rpath,$with_flint/lib $LDFLAGS"
-     CPPFLAGS="-I$with_flint/include $CPPFLAGS"
-   ])
-AC_ARG_ENABLE([flint],
-    [AS_HELP_STRING([--enable-flint@<:@=ARG@:>@],
-      [enable flint @<:@default=check@:>@])],
-    [:],
-    [enable_flint=check])
-AS_IF([test "x$enable_flint" != xno],
-    [FLINT_LIBS="-lflint -lmpfr"
-     LIBS_SAVED="$LIBS"
-     LIBS="$LIBS $FLINT_LIBS"
-     AC_CHECK_LIB(flint,dummy_fn_to_prevent_ISOC_warning,[have_flint=yes],[have_flint=no])
-     LIBS="$LIBS_SAVED"
-    ])
-AS_IF([test x$have_flint = xyes],
-  [enable_flint=yes],
-  [AS_IF([test "x$enable_flint" = xyes],
-    [AC_MSG_ERROR([flint is not available])],
-    [AS_IF([test "x$enable_flint" != xno],
-        [AC_MSG_NOTICE([flint will not be used because flint could not be found])])
-     enable_flint=no])])
+   [ AS_HELP_STRING([--with-flint=DIR],
+      [provide location of flint installation prefix or source directory]) ],
+   [],
+   [with_flint=check]
+)
+AS_IF([test "x$with_flint" != xno],
+    [ FLINT_LIBS="-lflint -lmpfr"
+    AS_CASE(["x$with_flint"],
+      [xyes|xcheck|x],[],
+      [*],
+      [ FLINT_LDFLAGS="-L$with_flint/lib -Wl,-rpath,$with_flint/lib $LDFLAGS"
+        FLINT_CPPFLAGS="-I$with_flint/include $CPPFLAGS" ] )
+      AX_CHECK_LIBRARY([FLINT],[flint/flint.h],[flint],fmpz_poly_cyclotomic,[have_flint=yes],[have_flint=no],[-lmpfr])
+    AS_IF([test x$have_flint = xyes],
+      [enable_flint=yes
+      LDFLAGS=$FLINT_LDFLAGS $LDFLAGS
+      CPPFLAGS=$FLINT_CPPFLAGS $CPPFLAGS ],
+      [AS_IF([test "x$with_flint" != xcheck],
+        [AC_MSG_ERROR([flint is not available])],
+        [AC_MSG_NOTICE([flint will not be used because flint could not be found])
+        enable_flint=no])
+      ]
+    )
+])
 AC_SUBST(FLINT_LIBS)
-AC_SUBST(COCOA_CPPFLAGS)
 AM_CONDITIONAL(NMZ_FLINT, test "x$enable_flint" = xyes)
 AS_IF([test "x$enable_flint" = xyes],
   [AC_DEFINE(NMZ_FLINT)])

--- a/m4/ax_check_library.m4
+++ b/m4/ax_check_library.m4
@@ -1,0 +1,96 @@
+# ===========================================================================
+#     https://www.gnu.org/software/autoconf-archive/ax_check_library.html
+# ===========================================================================
+#
+# SYNOPSIS
+#
+#   AX_CHECK_LIBRARY(VARIABLE-PREFIX, HEADER-FILE, LIBRARY-FILE, FUNCTION,
+#                    [ACTION-IF-FOUND], [ACTION-IF-NOT_FOUND], [other-libraries])
+#
+# DESCRIPTION
+#
+#   Provides a generic test for a given library, similar in concept to the
+#   PKG_CHECK_MODULES macro used by pkg-config.
+#
+#   Most simple libraries can be checked against simply through the
+#   presence of a header file and a library to link to. This macro allows to
+#   wrap around the test so that it doesn't have to be recreated each time.
+#
+#   Rather than define --with-$LIBRARY arguments, it uses variables in the
+#   same way that PKG_CHECK_MODULES does. It doesn't, though, use the same
+#   names, since you shouldn't provide a value for LIBS or CFLAGS but rather
+#   for LDFLAGS and CPPFLAGS, to tell the linker and compiler where to find
+#   libraries and headers respectively.
+#
+#   If the library is found, HAVE_PREFIX is defined, and in all cases
+#   PREFIX_LDFLAGS and PREFIX_CPPFLAGS are substituted.
+#
+#   Example:
+#
+#     AX_CHECK_LIBRARY([LIBEVENT], [event.h], [event], [main], [],
+#                      [AC_MSG_ERROR([Unable to find libevent])])
+#
+# LICENSE
+#
+#   Copyright (c) 2010 Diego Elio Petteno` <flameeyes@gmail.com>
+#
+#   This program is free software: you can redistribute it and/or modify it
+#   under the terms of the GNU General Public License as published by the
+#   Free Software Foundation, either version 3 of the License, or (at your
+#   option) any later version.
+#
+#   This program is distributed in the hope that it will be useful, but
+#   WITHOUT ANY WARRANTY; without even the implied warranty of
+#   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+#   Public License for more details.
+#
+#   You should have received a copy of the GNU General Public License along
+#   with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+#   As a special exception, the respective Autoconf Macro's copyright owner
+#   gives unlimited permission to copy, distribute and modify the configure
+#   scripts that are the output of Autoconf when processing the Macro. You
+#   need not follow the terms of the GNU General Public License when using
+#   or distributing such scripts, even though portions of the text of the
+#   Macro appear in them. The GNU General Public License (GPL) does govern
+#   all other use of the material that constitutes the Autoconf Macro.
+#
+#   This special exception to the GPL applies to versions of the Autoconf
+#   Macro released by the Autoconf Archive. When you make and distribute a
+#   modified version of the Autoconf Macro, you may extend this special
+#   exception to the GPL to apply to your modified version as well.
+
+#serial 5
+
+AC_DEFUN([AX_CHECK_LIBRARY], [
+  AC_ARG_VAR($1[_CPPFLAGS], [C preprocessor flags for ]$1[ headers])
+  AC_ARG_VAR($1[_LDFLAGS], [linker flags for ]$1[ libraries])
+
+  AC_CACHE_VAL(AS_TR_SH([ax_cv_have_]$1),
+    [save_CPPFLAGS="$CPPFLAGS"
+     save_LDFLAGS="$LDFLAGS"
+     save_LIBS="$LIBS"
+
+     AS_IF([test "x$]$1[_CPPFLAGS" != "x"],
+       [CPPFLAGS="$CPPFLAGS $]$1[_CPPFLAGS"])
+
+     AS_IF([test "x$]$1[_LDFLAGS" != "x"],
+       [LDFLAGS="$LDFLAGS $]$1[_LDFLAGS"])
+
+     AC_CHECK_HEADER($2, [
+       AC_CHECK_LIB($3, [$4],
+         [AS_TR_SH([ax_cv_have_]$1)=yes],
+         [AS_TR_SH([ax_cv_have_]$1)=no],
+         [$7])
+     ], [AS_TR_SH([ax_cv_have_]$1)=no])
+
+     CPPFLAGS="$save_CPPFLAGS"
+     LDFLAGS="$save_LDFLAGS"
+     LIBS="$save_LIBS"
+    ])
+
+  AS_IF([test "$]AS_TR_SH([ax_cv_have_]$1)[" = "yes"],
+    AC_DEFINE([HAVE_]$1, [1], [Define to 1 if ]$1[ is found])
+    [$5],
+    [$6])
+])

--- a/source/Makefile.am
+++ b/source/Makefile.am
@@ -19,6 +19,7 @@ LIBS += $(COCOA_LIBS)
 endif
 if NMZ_FLINT
 LIBS += $(FLINT_LIBS)
+
 endif
 LIBS += -lgmpxx -lgmp -lpthread
 

